File: day-3/part-1.pl
#!/usr/bin/perl
open(my $in, "<", "day-3/input");
my @lines = <$in>;

my $repeats = ($#lines + 1) * 3;

my $i = 0;
my $trees = 0;

foreach my $line (@lines) {
    $line =~ s/\R//g;
    $line = "$line" x $repeats;
    my $char = substr($line, $i, 1);
    if ( "$char" eq "#" ) {
        $trees = $trees + 1;
    }
    $i = $i + 3;
}

print("$trees\n");

File: day-3/part-1.py
import os
dirname = os.path.dirname(__file__)
filename = os.path.join(dirname, 'input')

input = open(filename, "r")

with open(filename) as f:
   count = sum(1 for _ in f)

repeats = count * 3

i = 0
trees = 0

for line in input.readlines():
    line = line.strip()
    line = line * repeats
    c = line[i]
    if c == "#":
        trees += 1
    i += 3

print(trees)
